From c237e59fc7899ccb5ef7ac4f2e5859ca01425402 Mon Sep 17 00:00:00 2001
From: Zhang Xiao <xiao.zhang@windriver.com>
Date: Tue, 24 Mar 2015 15:54:49 +0800
Subject: [PATCH 3/3] fs/yaffs2: Fix a judgement logic for ACL operations

Currently yaffs doesn't support ACL thus all related
operations should return error. While the judgement
on ACL operation miss a "!" thus cause this mechanism
failed. Add a "!" to fix it.

Signed-off-by: Zhang Xiao <xiao.zhang@windriver.com>
---
 fs/yaffs2/yaffs_vfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/yaffs2/yaffs_vfs.c b/fs/yaffs2/yaffs_vfs.c
index a7a7d06..52177c1 100644
--- a/fs/yaffs2/yaffs_vfs.c
+++ b/fs/yaffs2/yaffs_vfs.c
@@ -932,7 +932,7 @@ static int yaffs_setxattr(struct dentry *dentry, const char *name,
 	/* Currently we don't support posix ACL so never accept any settings
 	 * start with "system.posix_acl_".
 	 */
-	if (strncmp(name, "system.posix_acl_", 17))
+	if (!strncmp(name, "system.posix_acl_", 17))
 		error = -EOPNOTSUPP;
 
 	if (error == 0) {
-- 
2.0.2

